### ISSUE TITLE
[MenuItem] isFocusable prop is undocumented

### ISSUE THREAD

**Issue by ljosberinn** [chakra-ui/chakra-ui#1651]
...and does not work when used together with `isDisabled`, although there is a story for it (which also doesn't work).

We should either fix & document or consider dropping it. 

relevant code:

https://github.com/chakra-ui/chakra-ui/blob/develop/packages/menu/src/use-menu.ts#L431
https://github.com/chakra-ui/chakra-ui/blob/develop/packages/menu/stories/menu.stories.tsx#L102

**Comment by ljosberinn** [comment]
`MenuItemOption` has `isChecked` but it's always only shown within a `MenuOptionGroup` and does not seem to work if within one. It's also undocumented. So I guess this is a typing mistake [here](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/menu/src/menu.tsx#L241) because of extending `UseMenuOptionOptions`.

**Comment by TimKolberger** [comment]
I looked into this and learned some things:

> Elements with `aria-disabled="true"` cannot receive focus. 
> – [MDN Attributes disabled](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/disabled#:~:text=Attribute%20interactions,controls%20until%20they%20are%20enabled.)

The combination of `isDisabled` and `isFocusable` is not valid and I think we should not just "simply fix the bug".

I found three ways to go:

1. We could use the `aria-readonly` property to emulate `isDisabled && isFocusable`
    * Needs changes in `useClickable`
2. Define `isFocusable` with a more narrow type: `isFocusable: undefined | false` and give it the default value `isFocusable = !isDisabled`
3. Remove `isFocusable` completely

Which one is the best to go?
